modules/autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/ui/ExistingCredentialMatchDetector.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.autofill.ui
+
+import com.duckduckgo.autofill.store.AutofillStore
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesTo
+import dagger.Module
+import dagger.Provides
+
+class ExistingCredentialStoreInterrogatingMatchDetector(private val autofillStore: AutofillStore) : ExistingCredentialMatchDetector {
+
+    override suspend fun determine(currentUrl: String, username: String, password: String): AutofillStore.ContainsCredentialsResult {
+        return autofillStore.containsCredentials(currentUrl, username, password)
+    }
+}
+
+@Module
+@ContributesTo(AppScope::class)
+object ExistingCredentialMatchDetectorStoreModule {
```

Comment:
NIT: Since these classes (a few other more I've noticed) are in the impl module, we should be able to use @ContributesBinding instead of creating a module